backend/models/devices.js

```diff
@@ -0,0 +1,63 @@
+let index = 3;
+
+let devices = {
+    device1: {
+        id: 'device1',
+        name: 'Device #1',
+        address: '192.168.1.50',
+        port: 90,
+        state: 'on'
+    },
+    device2: {
+        id: 'device2',
+        name: 'Device #2',
+        address: '192.168.1.60',
+        port: 80,
+        state: 'off'
+    }
+};
+
+function getAllDevices() {
+    return Object.values(devices);
+}
+
+function getDeviceById(deviceId) {
+    return devices[deviceId];
+}
+
+function addDevice(device) {
+    index += 1;
+    devices['device' + index] = {
+        id: 'device' + index,
+        state: 'off',
+        ...device
+    };
+}
+
+function deleteDevice(id) {
+    devices = {
+        ...devices,
+        [id]: undefined
+    };
+
+    delete devices[id];
```

Comment:
Why resetting it to undefined before deleting?
You're not trying to make it immutable or anything, why lines 38..41?